sphinx_gallery/gen_rst.py

```diff
@@ -565,6 +565,13 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     time_elapsed = 0
     block_vars = {'execute_script': execute_script, 'fig_count': 0,
                   'image_path': image_path_template, 'src_file': src_file}
+
+    argv_orig = sys.argv[:]
+    if block_vars['execute_script']:
+        print('Executing file %s' % src_file)
+        sys.argv[0] = src_file
```

Comment:
Otherwise `sys.argv[0]` is `sphinx-build` leading to a very confusing error message like the one shown in https://github.com/sphinx-gallery/sphinx-gallery/issues/251#issue-236174916.